api/src/Repository/Reaction.hs

```diff
@@ -20,3 +22,33 @@ createReaction reaction = do
 
 getReactionsByPipelineId' :: PipelineId -> AppM [Reaction Identity]
 getReactionsByPipelineId' pId = runQuery (select $ orderBy (reactionOrder >$< asc) $ getReactionsByPipelineId pId)
+
+getWorkflow' :: PipelineId -> AppM (Pipeline Identity, [Reaction Identity])
+getWorkflow' pId = do
+  res <- runQuery $ select $ limit 1 $ getWorkflow pId
+  return $ head res
+
+getWorkflows' :: AppM [(Pipeline Identity, [Reaction Identity])]
+getWorkflows' = runQuery $ select getWorkflows
+
+getWorkflowsByUser' :: UserId -> AppM [(Pipeline Identity, [Reaction Identity])]
+getWorkflowsByUser' uid = runQuery $ select $ getWorkflowsByUser uid
+
+deleteReactionsByPipelineId' :: PipelineId -> AppM Int64
+deleteReactionsByPipelineId' pId = runQuery $ delete $ deleteReactionsByPipelineId pId
+
+createReactions :: [Reaction Identity] -> AppM [ReactionId]
+createReactions = mapM createReaction
+
+putWorkflow :: PipelineId -> Pipeline Expr -> [Reaction Identity] -> AppM Int64
+putWorkflow pId newPipeline newReactions = do
+  row <- runQuery $ update $ updatePipeline pId newPipeline
+  reactionsDeleted <- deleteReactionsByPipelineId' pId
```

Comment:
You should use the CASCADE delete option of postgres instead of creating your own custom request.